src/Storages/StorageDistributed.cpp

```diff
@@ -1797,6 +1797,26 @@ void StorageDistributed::flushClusterNodesAllData(ContextPtr local_context)
         node->flushAllData();
 }
 
+void StorageDistributed::flushClusterNodesAllSettings(ContextPtr local_context)
```

Comment:
This is almost 100% copy of `StorageDistributed::flushClusterNodesAllData`, I suggest factoring out populating `directory_monitors` into a separate function and re-using it.

Reply:
ok